src/Assert/Assert.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Assert;
+
+use Wikimedia\Assert\Assert as WikimediaAssert;
+use Wikimedia\Assert\ParameterAssertionException;
+
+/**
+ * Provides functions to assure values meet certain preconditions relevant
+ * in Wikibase.
+ *
+ * @see Wikimedia\Assert\Assert
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class Assert {
+
+	/**
+	 * @param string $value
+	 * @param string $name
+	 *
+	 * @throws ParameterAssertionException if $value is not an allowable repository name.
+	 */
+	public static function parameterIsAllowableRepositoryName( $value, $name ) {
```

Comment:
Yeah, names. So I initially also thought about "valid" but it also felt like "valid repository name" could be understood as the "repository name of a foreign repository that has been configured" etc. while this is just about a name format.
But yeah, maybe it is over-thinking things. Will change to "assertParameterIsValidRepositoryName".